js/ui/components/clubhouse.js

```diff
@@ -0,0 +1,323 @@
+        if (!this._useClubhouse) {
+            log('Cannot enable Clubhouse in this image version');
+            return;
+        }
+
+        log('Enabling Clubhouse component');
+
+        this._clubhouseProxy = new Gio.DBusProxy.new_sync(sessionBus,
+                                                          Gio.DBusProxyFlags.DO_NOT_AUTO_START_AT_CONSTRUCTION,
+                                                          null,
+                                                          CLUBHOUSE_ID,
+                                                          CLUBHOUSE_DBUS_OBJ_PATH,
+                                                          CLUBHOUSE_ID,
+                                                          null);
+        this._clubhouseProxy.connect('notify::g-name-owner', Lang.bind(this, function() {
+            if (!this._clubhouseProxy.g_name_owner) {
+                log('Nothing owning D-Bus name %s, so dismiss the Clubhouse banner'.format(CLUBHOUSE_ID));
+                this._dismiss_quest_cb();
+            }
+        }));
+
+        this.actor = new St.Widget({ visible: true,
+                                     clip_to_allocation: true,
+                                     can_focus: true,
+                                     track_hover: true,
+                                     reactive: true,
+                                     layout_manager: new Clutter.BinLayout() });
+
+        Main.layoutManager.addChrome(this.actor);
+
+        this.source = new ClubhouseNotificationSource(CLUBHOUSE_ID);
+        this.source.connect('notify', Lang.bind(this, this._onNotify));
+
+        this._overrideAddNotification();
+    },
+
+    disable: function() {
+        if (!this._useClubhouse)
+            return;
+
+        log('Disabling Clubhouse component');
+
+        if (this.actor)
+            this.actor.destroy();
+
+        if (this._clubhouseProxyHandler > 0)
+            this._clubhouseProxy.disconnect(this._clubhouseProxyHandler);
+
+        if (this._oldAddNotificationFuncPrototype) {
+            NotificationDaemon.GtkNotificationDaemon.prototype.AddNotificationAsync =
+                this._oldAddNotificationFuncPrototype;
+        }
+
+        this.actor = null;
+        this.source = null;
+        this._banner = null;
+        this._clubhouseProxy = null;
+    },
+
+    _imageUsesClubhouse: function(eosImageVersion) {
+        // We are only using the Clubhouse component in Endless Hack images for now, so
+        // check if the prefix of the image version matches the mentioned flavor.
+        return eosImageVersion.startsWith('hack-');
+    },
+
+    _overrideAddNotification: function() {
+        this._oldAddNotificationFuncPrototype =
+            NotificationDaemon.GtkNotificationDaemon.prototype.AddNotificationAsync;
+        NotificationDaemon.GtkNotificationDaemon.prototype.ClubhouseData = [CLUBHOUSE_ID,
+                                                                            this.source];
```

Comment:
I tried that in the beginning but it was failing when destroying the existing source for the app ID (because the messageTray's code runs before the components one).
Maybe it had to do with the fact that I had saved notifications, but in any case it seemed to be a safer way to do this.